### ISSUE TITLE
Custom contenttype is not actually attached to the list when using SPFx asset provisioning

### ISSUE THREAD

**Issue by justdevelopment** [SharePoint/sp-dev-docs#1518]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior
Custom ContentType should be attached to a newly create list on asset creation or if not possible, added after it has become available to the list through some sort of timerjob? 

#### Observed Behavior
The list is created, all columns are added according to ContentType but the ContentType itself is not added to the list. It has been created within the site collection but also cannot be added manually (unavailable in options) to the list. After some time the ContentType is suddenly available in the settings and can be added.

#### Steps to Reproduce
-	Create a SPFx webpart. I’ve made a really simple demo based on the [docs ](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/get-started/provision-sp-assets-from-package)here; [Github repo](https://github.com/justdevelopment/SPFxAssetDemo
)
-	Upload the app as you would normally and install it in the web.
-	After it has been installed on the web, see that the list has been created and contains all columns. 
-	When you add a new item or go to list settings; see that ContentType has been enabled but only ‘Item’ is there.

![image](https://user-images.githubusercontent.com/23448165/37774784-7552540e-2de1-11e8-82c1-1af0f3b84ca0.png)

-	Go to Site Settings and check the available ContentTypes; see that our custom one is added to the group 'SPFx Content Types'
![image](https://user-images.githubusercontent.com/23448165/37775040-16d22868-2de2-11e8-80d8-ae53203168d6.png)

-	Go to list settings and choose to add an existing ContentType
-	See that our new custom ContentType is not available for selection in the settings (notice that the new group is also unavailable).
![image](https://user-images.githubusercontent.com/23448165/37775095-4c8a7cd0-2de2-11e8-877d-1ff536d53bd0.png)

-	Wait 30-60 min, repeat above steps and now you can add the new ContentType to the list.

**Comment by justdevelopment** [comment]
It is set to true: https://github.com/justdevelopment/SPFxAssetDemo/blob/master/sharepoint/assets/schema.xml

ContentTypes have been enabled for the list, it's just not adding a custom one.

**Comment by mcmynn83** [comment]
@justdevelopment - thanks for the response, we will take a look at this as it seems like an issue.

**Comment by balle3201** [comment]
Hi guys. Any news on this issue? I am trying to accomplish the same in another solution, but am still left with only "Item" as the single available Content Type. The fields however, are provisioned as expected.

**Comment by UX-Gav** [comment]
@justdevelopment / @balle3201 
Ok well I don't have issues. Item has been extended. In list settings i also only see "Item". What you are seeing is the extension of Item through your Content Type. And because there's an Item on every list, Item becomes Item. Inheritance right?

**Click on Item  >>**
![listsetting](https://user-images.githubusercontent.com/28395947/40792433-3a2e92ac-64fa-11e8-93e2-cfd7e5badc67.jpg)


This screen now shows the extended Item, inherited through your custom content type which inherits from the Site Content Type: Item. Notice the **Parent** is your custom content type. In my case I called it "Footer". 

![listcontenttype](https://user-images.githubusercontent.com/28395947/40793499-d9a64e4a-64fc-11e8-8077-8775feb7775e.jpg)

Now **Click** on the parent content type at the top ** (For me Footer) 
and you get:

![boom](https://user-images.githubusercontent.com/28395947/40792471-4ad90506-64fa-11e8-89e1-e7a2b5b286cd.jpg)

and there she is.. inheriting from Item and passing that down and so on...  works for me and makes sense. 

## Basically:

 -  **Item** ( Site Content Type )

    -  _## **Footer** ( Custom Content Type )_

       -  **Item** ( List Content Type )

           -  Custom List ( inherits from above )

It's just like normal, except that your Custom Content Type has been inserted between Item (Site) and Item (List) as above. 

Inheritance of columns from **Item** as well as your **Custom Content Type's ** associated columns.

I think you assumed that when you saw **Item** (List Content Type) that it inherited from **Item** (Site Content Type) as per normal when you manually create a new custom list in the UI. 
You were probably expecting to see your Custom Content Type below Item in List Settings.

Hope this helps clarify? If not, apologies in advance for the spam :)

**Comment by justdevelopment** [comment]
You're right, it does seem to inherit from the created contenttype (nicely spotted!), but that still isn't expected behavior. We want it to be the contenttype we create by name, not as 'Item'. Plus, it's weird that when you manually add the newly created contenttype you can only do so after X minutes and then it works exactly as expected and as in previous SharePoint versions. 

I haven't done any further testing yet, but I am wondering now what will happen if you add two custom contenttypes. Will all columns be added to Item, how will it inherit then? Or do you get two 'Item' contenttypes? Did you perhaps test this scenario? If not, we will do some tests to find out :)

**Comment by UX-Gav** [comment]
Hi. Haven't tested the scenario with multiple content types yet. I'll let you have a bash and let us know :)

**Comment by mohitcreative** [comment]
I can still replicate this issue while provisioning assets. Was the fix made or is it the itended behaviour that we will see Content Type name as "Item" derived from our custom content type "x". I have the exact situation as @UX-Gav has explained.

**Comment by justdevelopment** [comment]
I completely forgot about this ticket... Oops?

So we did manage to figure out a way to get it to work correctly: use the entire ContentType in your listschema, not the ContentTypeRef.

So [here in this file ](https://github.com/justdevelopment/SPFxAssetDemo/blob/master/sharepoint/assets/schema.xml) we have a ContentTypeRef: 
```xml
<ContentTypeRef ID="0x010042D0C1C200A14B6887742B6344675C8B" />
```

It should actually be this instead:
```XML
 <ContentType ID="0x010042D0C1C200A14B6887742B6344675C8B"
           Name="Cost Center"
           Group="SPFx Content Types"
           Description="Sample content types from web part solution">
       <FieldRefs>
           <FieldRef ID="{060E50AC-E9C1-4D3C-B1F9-DE0BCAC300F6}" />
           <FieldRef ID="{943E7530-5E2B-4C02-8259-CCD93A9ECB18}" />
       </FieldRefs>
   </ContentType>
```

Now your list will get created with the proper contenttype attached, all fields as they should be and no 'Item' in sight.

**Comment by mohitcreative** [comment]
Do I have to make a corresponding change in Elements.XML as well? After making your suggested change, I have same XML structure of ContentType tag both in Elements.XML and Schema.XML

![CT](https://user-images.githubusercontent.com/8166231/56035379-03636080-5cf8-11e9-9718-f0aa2d2c20fb.jpg)

**Comment by justdevelopment** [comment]
That's correct, you will have the same in both files. This way you will get
the site content type and a list content type that inherits from that.

**Comment by mohitcreative** [comment]
Thank you. It worked. If anyone faces this situation, I will post my complete solution in a while and provide the link here.
